app/models/service.rb

```diff
@@ -427,4 +428,22 @@ def remove_from_service(parenent_service)
     update(:parent => nil)
     parenent_service.remove_resource(self)
   end
+
+  private
+
+  def apply_dialog_settings
+    dialog_options = options[:dialog] || {}
+
+    %w(dialog_service_name dialog_service_description).each do |field_name|
+      send(field_name, dialog_options[field_name]) if dialog_options.key?(field_name)
+    end
+  end
+
+  def dialog_service_name(value)
+    self.name = value if value.present?
```

Comment:
@gmcculloug 
Should we allow for designers to append a date_time to the service using something like
My_Service_%{date_time}
And we would put the current date_time in the name field.

Reply:
the designer could leave the field as readonly so the end user ordering it doesn't have to set the service name.

Reply:
Users can do this today through automate, I would rather not introduce it here yet.

Reply:
@gmcculloug 
From a usability perspective the service names are all going to be same out of the box. If the user provisions the same service multiple times they would all have the same name. And for services that don't use automate they won't have a way to discern one from the other.